braintree-to-redshift/braintree_connection.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+""" Functions that connect to Braintree and pull transaction and dispute data.
+"""
+
+import sys
+import os
+import re
+import braintree
+import decimal
+from datetime import date, datetime, timedelta
+from time import gmtime, strftime, time
+import json
+from settings import (braintree_merchant_id, braintree_public_key, braintree_private_key, braintree_env)
+
+local_settings_path = os.path.join(os.getcwd(),"settings.py")
+if os.path.exists(local_settings_path):
+    import imp
+    settings = imp.load_source('settings', local_settings_path)
+
+
+def decimal_default(obj):
+    if isinstance(obj, decimal.Decimal):
+        return float(obj)
+    raise TypeError
+
+def datetime_default(o):
+    if isinstance(o, datetime):
+        return o.__str__()
+
+def connect_to_braintree():
+    gateway = braintree.BraintreeGateway(
+        braintree.Configuration(
+            environment=braintree.Environment.Sandbox,
+            merchant_id=braintree_merchant_id,
+            public_key=braintree_public_key,
+            private_key=braintree_private_key,
+        )
+    )
+    return gateway
+
+def create_transaction(amount, nonce):
+    gateway = connect_to_braintree()
+    result = gateway.transaction.sale({
+        "amount": amount,
+        "payment_method_nonce": nonce,
+        "options": {
+            "submit_for_settlement": True
+        }
+    })
+
+    if result.is_success:
+        print("success!: " + result.transaction.id)
+    elif result.transaction:
+        print("Error processing transaction:")
+        print("  code: " + result.transaction.processor_response_code)
+        print("  text: " + result.transaction.processor_response_text)
+    else:
+        for error in result.errors.deep_errors:
+            error_string = "attribute: %s \tcode: %s \t message: %s" % (error.attribute, error.code, error.message)
+            log_error(error_string, './errors.txt')
```

Comment:
if we run this on Lambda, then these errors won't be visible because files go away with the ephemeral virtual environment.
We've often succumbed just to `print` statements for errors which are visible in Lambda logs in AWS.

Idiomatically, python generally uses the core [logging module](https://docs.python.org/3/howto/logging-cookbook.html)